recipes/sipbuildtool/conanfile.py

```diff
@@ -0,0 +1,140 @@
+import os
+import sys
+
+from pathlib import Path
+from io import StringIO
+
+from jinja2 import Template
+
+from conan import ConanFile
+from conan.tools.env import VirtualRunEnv
+from conan.tools.files import files
+from conans.errors import ConanException
+
+
+class SipBuildTool(object):
+    """
+    A generator that creates a CMake script to build python bindings for
+    projects using Sip.
+
+    This generator expects the project to have a pyproject.toml.jinja file that
+    determines which Sip files to compile. It configures that file to a TOML
+    file that Sip can use.
+
+    Usage in a conanfile for a Sip project:
+    sip = self.python_requires["sipbuildtool"].module.SipBuildTool(self)
+    sip.configure()
+    sip.generate("projectName")
+    """
+    def __init__(self, conanfile: ConanFile):
+        self.conanfile = conanfile
+        env = VirtualRunEnv(self.conanfile)
+        self.run_env = env.environment()
+        self.envvars = None
+        self.python_venv_interpreter = None
+        self.sip_build_executable = None
+
+    @property
+    def _venv_base_path(self):
+        return os.path.join(self.conanfile.build_folder, "venv")
+
+    @property
+    def _venv_bin_path(self):
+        if self.conanfile.settings.os == "Windows":
+            return os.path.join(self._venv_base_path, "Scripts")
+        return os.path.join(self._venv_base_path, "bin")
+
+    def configure(self, sip_version = "6.5.1", python_interpreter = sys.executable):
+        python_interpreter = Path(python_interpreter)
+
+        # When on Windows execute as Windows Path
+        if self.conanfile.settings.os == "Windows":
+            python_interpreter = Path(*[f'"{p}"' if " " in p else p for p in python_interpreter.parts])
+
+        # Create the virtual environment
+        self.conanfile.run(f"""{python_interpreter} -m venv {self._venv_base_path}""", env = "conanbuild")
+
+        # Make sure there executable is named the same on all three OSes this allows it to be called with `python`
+        # simplifying GH Actions steps
+        if self.conanfile.settings.os != "Windows":
+            self.python_venv_interpreter = Path(self.conanfile.build_folder, self._venv_bin_path, "python")
+            if not self.python_venv_interpreter.exists():
+                self.python_venv_interpreter.hardlink_to(Path(self.conanfile.build_folder, self._venv_bin_path,
+                                                              Path(sys.executable).stem + Path(sys.executable).suffix))
+        else:
+            self.python_venv_interpreter = Path(self.conanfile.build_folder, self._venv_bin_path,
+                                                Path(sys.executable).stem + Path(sys.executable).suffix)
```

Comment:
Why not use the `name` property here?

```suggestion
                                                Path(sys.executable).name)
```

https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.name it seems to do the same as https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem + https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffix